src/providers/python.rs

```diff
@@ -0,0 +1,37 @@
+use anyhow::Result;
+
+use crate::{providers::Pkg, nixpacks::app::App};
+
+use super::Provider;
+
+pub struct PythonProvider {}
+impl Provider for PythonProvider {
+    fn name(&self) -> &str {
+        "python"
+    }
+
+    fn detect(&self, app: &crate::nixpacks::app::App) -> anyhow::Result<bool> {
+        Ok(app.includes_file("main.py"))
```

Comment:
Whilst most apps have this as their entry point, not all apps do. It should look for any python file.

Reply:
Not well-versed with Python but might make more sense to check for a `requirements.txt` file?

Reply:
`requirements.txt` doesn't have a main-file entry. That's why I suggested checking for all python files and seeing if there is one that makes sense to run. This could be heavily inspired by the Heroku one: https://github.com/heroku/heroku-buildpack-python

Reply:
+1 to using Heroku buildpack as reference for detection. It looks like it checks for the exist of the 3 files and if any exist considers it a Python project.

![image](https://user-images.githubusercontent.com/3044853/161570281-ad4740ad-fa03-4d77-8f6e-48e0c869d325.png)